src/statusbar/targetStatusBar.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  StatusBarItem,
+  StatusBarAlignment,
+  window,
+  Disposable,
+  TextEditor,
+  workspace,
+  Uri,
+} from "vscode";
+import { provideSingleton } from "../utils";
+import { DBTProjectContainer } from "../manifest/dbtProjectContainer";
+import { DBTTerminal } from "../dbt_client/dbtTerminal";
+
+@provideSingleton(TargetStatusBar)
+export class TargetStatusBar implements Disposable {
+  readonly statusBar: StatusBarItem = window.createStatusBarItem(
+    StatusBarAlignment.Left,
+    9,
+  );
+  private disposables: Disposable[] = [];
+
+  constructor(
+    private dbtProjectContainer: DBTProjectContainer,
+    private dbtTerminal: DBTTerminal,
+  ) {
+    this.disposables.push(
+      window.onDidChangeActiveTextEditor(
+        async (event: TextEditor | undefined) => {
+          this.updateStatusBar();
+        },
+      ),
+    );
+  }
+
+  dispose() {
+    while (this.disposables.length) {
+      const x = this.disposables.pop();
+      if (x) {
+        x.dispose();
+      }
+    }
+    this.statusBar.dispose();
+  }
+
+  public async updateStatusBar() {
+    try {
+      const currentProject = this.getCurrentProject();
+      const targets = await currentProject.getTargetNames();
+      const selectedTarget = await currentProject.getSelectedTarget();
+      this.statusBar.text = `$(target) ${selectedTarget}`;
+      this.statusBar.command = {
+        title: "Open Target selector",
+        command: "dbtPowerUser.openTargetSelector",
+        arguments: [targets, currentProject, this],
+      };
+      this.statusBar.show();
+    } catch (err) {
+      this.statusBar.hide();
+      this.dbtTerminal.debug(
+        "TargetStatusBar",
+        "Unable to update target status bar",
+        err,
+      );
+    }
+  }
+
+  private getCurrentProject() {
```

Comment:
i think @saravmajestic created a helper util to get current project in `src/service/queryManifestService.ts`

Reply:
it is not exactly the same. This picks the single project even if no text editor is open.

Reply:
We should probably also move this logic to this helper service though.